lib/service/database.js

```diff
@@ -418,6 +449,12 @@ function setCacheQueryTimeoutFromLimits(node, limits) {
     });
 }
 
+function getSourceNodes(node) {
```

Comment:
I feel like this function is pretty useful, it could be accessible from Node, couldn't it?